fossunited/foss_profiles/doctype/foss_user_profile/test_foss_user_profile.py

```diff
@@ -1,8 +1,47 @@
 # Copyright (c) 2023, Frappe x FOSSUnited and Contributors
 # See license.txt
 
+import uuid
+
+import frappe
 from frappe.tests.utils import FrappeTestCase
 
 
 class TestFOSSUserProfile(FrappeTestCase):
-    pass
+    def test_add_profile(self):
+        # Given a user that does not have a FOSSUnitedProfile
+        inserted_username = "wisharya"
+        inserted_name = "wish arya"
+        profile_exists = frappe.db.exists(
+            "FOSS User Profile", {"username": inserted_username}
+        )
+        self.assertFalse(profile_exists)
+
+        # When a FOSSUnitedProfile is created for the user
+        # Note that a Frappe User needs to exist before a Profile can be created
+        # Verify that only required values are provided below
+        frappe_user = frappe.get_doc(
+            {
+                "doctype": "User",
+                "username": inserted_username,
+                # Create a unique email address as it is used as a database key
+                "email": str(uuid.uuid4()) + "@fossunited.org",
+                "first_name": "wish",
+                "name": inserted_name,
+                "full_name": inserted_name,
+            },
+        ).insert()
```

Comment:
This step must be creating a foss user on it's own. 
Ref: [user_utils.py file](https://github.com/fossunited/fossunited/blob/develop/fossunited/fossunited/user_utils.py) , function `create_profile_on_user_create`
Also
[Hooks file](https://github.com/fossunited/fossunited/blob/develop/fossunited/hooks.py) line 181.


This is why tests are failing on next line with duplicate entry error. A profile already exists with this user linked to it.

Suggestion: don't do line 34 to 41, instead just check if profile exists that is linked to the user.

```py
profile_exists = frappe.db.exists(
    "FOSS User Profile", {"user": frappe_user.name}
)
```

Reply:
perfect. thanks for debugging this @harshtandiya ! I can only make the changes on Friday now. @Mangeshrex @harshtandiya  any chance you could pick this up and address it tomorrow? If not, I will address it on Friday.

Reply:
The tests seems to pass when I'm trying to do what harsh has suggested. 
![Capture-Jul-11-18-54-01](https://github.com/fossunited/fossunited/assets/82205152/57a3b244-3dd3-4c01-8440-a7d502191749)